src/pages/iou/NewDistanceRequestPage.tsx

```diff
@@ -1,63 +1,42 @@
-import lodashGet from 'lodash/get';
-import PropTypes from 'prop-types';
+import type {StackScreenProps} from '@react-navigation/stack';
 import React, {useCallback, useEffect} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
 import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
 import DistanceRequest from '@components/DistanceRequest';
 import Navigation from '@libs/Navigation/Navigation';
-import reportPropTypes from '@pages/reportPropTypes';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
 import * as IOU from '@userActions/IOU';
-import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import {iouPropTypes} from './propTypes';
+import type SCREENS from '@src/SCREENS';
+import type {IOU as IOUType, Report} from '@src/types/onyx';
 
-const propTypes = {
+type NewDistanceRequestPageOnyxProps = {
     /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
-    iou: iouPropTypes,
+    iou: OnyxEntry<IOUType>;
 
     /** The report on which the request is initiated on */
-    report: reportPropTypes,
-
-    /** Passed from the navigator */
-    route: PropTypes.shape({
-        /** Parameters the route gets */
-        params: PropTypes.shape({
-            /** Type of IOU */
-            iouType: PropTypes.oneOf(_.values(CONST.IOU.TYPE)),
-            /** Id of the report on which the distance request is being created */
-            reportID: PropTypes.string,
-        }),
-    }),
+    report: OnyxEntry<Report>;
 };
 
-const defaultProps = {
-    iou: {},
-    report: {},
-    route: {
-        params: {
-            iouType: '',
-            reportID: '',
-        },
-    },
-};
+type NewDistanceRequestPageProps = NewDistanceRequestPageOnyxProps & StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.DISTANCE>;
 
 // This component is responsible for getting the transactionID from the IOU key, or creating the transaction if it doesn't exist yet, and then passing the transactionID.
 // You can't use Onyx props in the withOnyx mapping, so we need to set up and access the transactionID here, and then pass it down so that DistanceRequest can subscribe to the transaction.
-function NewDistanceRequestPage({iou, report, route}) {
-    const iouType = lodashGet(route, 'params.iouType', 'request');
+function NewDistanceRequestPage({iou, report, route}: NewDistanceRequestPageProps) {
+    const iouType = route.params.iouType;
```

Comment:
Why have you skipped the default value?
```suggestion
    const iouType = route.params.iouType ?? 'request';
```